bin/cask

```diff
@@ -324,10 +330,21 @@ def main():
             exit_error(
                 'Python 2.6 required, yours is {0.major}.{0.minor}'.format(
                     sys.version_info))
+
+
         if len(sys.argv) > 1 and sys.argv[1] == 'exec':
+            ## never returns!
             exec_command(sys.argv[2:])
-        else:
-            exec_cask(sys.argv[1:])
+
+        ## special handling for --emacs/bin as this has to happen before
+        ## everything else, and we don't want the parsing in two places.
+        if len(sys.argv) > 1:
+            if sys.argv[1] == '--emacs':
+                global EMACS
```

Comment:
> Is there a reason for get_cask_emacs -- it can be run several times (via get_cask_path) and each times requires a full Emacs invocation. Would it not make more sense to set this once an init time?

I'm not sure what you mean.  `get_cask_path` does not call `get_cask_emacs`.  It runs a new Cask subprocess—which in turn calls `get_cask_emacs` internally, for sure, but would you like to avoid that?

> Would you still be unhappy with the use of a global EMACS variable -- defined once in main, rather similar to the CASK* variables (which are also globals).

The CASK variables are global *constants*.  I'm curious how you'd implement a global `EMACS` variable—presumably storing the path to the selected Emacs instance—as a global constant… unless you move half of `main` into a function to initialize this constant, which I'd not like to see either.